src/transformers/models/t5/modeling_t5.py

```diff
@@ -2293,3 +2333,157 @@ def forward(
             encoder_hidden_states=encoder_outputs.hidden_states,
             encoder_attentions=encoder_outputs.attentions,
         )
+
+
+@add_start_docstrings(
+    """
+    T5 model utilizing mainly the encoder for sequence classification. Research has shown that this model can be more
+    efficient and usable over T5 and BERT for non-autoregressive tasks (EncT5: https://arxiv.org/abs/2110.08426).
+
+    This is similar to T5ForSequenceClassification, except: 1) there are less decoder layers (and as a result, less
+    parameters), 2) there is a separate decoder word embedding, and the decoder input ids are predefined constants, and
+    3) the classification head is designed differently (only projects, without using a FFN).
+    """,
+    T5_START_DOCSTRING,
+)
+class T5ForEncoderBasedSequenceClassification(T5PreTrainedModel):
+    _keys_to_ignore_on_load_unexpected = ["decoder.block.0.layer.1.EncDecAttention.relative_attention_bias.weight"]
+    _tied_weights_keys = ["transformer.encoder.embed_tokens.weight", "transformer.decoder.embed_tokens.weight"]
+
+    def __init__(self, config: T5Config):
+        super().__init__(config)
+
+        # Set number of decoders to 1.
+        single_decoder_layer_config = copy.deepcopy(config)
+        single_decoder_layer_config.num_decoder_layers = 1
+        self.transformer = T5Model(single_decoder_layer_config)
+
+        # Initiate decoder embedding from scratch and define the corresponding latent vector vocabulary size.
+        vocab_size = 1
+        if config.problem_type == "multi_label_classification":
+            vocab_size = config.num_labels
+        self.decoder_embeddings = nn.Embedding(vocab_size, config.d_model)
+        self.transformer.get_decoder().set_input_embeddings(self.decoder_embeddings)
```

Comment:
Thanks for pointing out the tie_encoder_decoder config flag. I think we should disable that somehow, and I will look into that.

However, I think we still need a new decoder embedding instance since the encoder's vocab_size is different from the decoder's vocab_size.